### ISSUE TITLE
Increase missions menu UI size

### ISSUE THREAD

**Issue by Kilvoctu** [CleverRaven/Cataclysm-DDA#78806]
### Is your feature request related to a problem? Please describe.

The missions UI is too small when on lower terminal size, making it difficult to read mission information.

### Solution you would like.

Ensure the UI is of a reasonable, usable size when on small terminal.

### Additional context

This is how the UI currently looks at a terminal size of 80x30
![Screenshot 2024-12-27 224036](https://github.com/user-attachments/assets/666c5822-7abb-46ba-a172-02a0cf2d314d)

It could be a simple 2-line fix to have the missions UI fill the screen...
![Screenshot 2024-12-27 225220](https://github.com/user-attachments/assets/8b506e2f-c94a-4fb3-9581-bfebdbe6898f)
..however, I see this commit 5ea3908e01314e55cfc7d7a75aa3cc3ab4d95b37 purposely reduced the size of the window, which I guess because someone didn't like dead space on larger terminals? 

I'm not sure how people would want to handle this.

**Comment by moxian** [comment]
> ..however, I see this commit https://github.com/CleverRaven/Cataclysm-DDA/commit/5ea3908e01314e55cfc7d7a75aa3cc3ab4d95b37 purposely reduced the size of the window, which I guess because someone didn't like dead space on larger terminals?

Yes, because mission window covering your entire screen is ugly when your screen is big. It stops being a "window" and looks bad. Current pocket settings behavior is a good example of this (and it looks bad).

> I'm not sure how people would want to handle this.

Come up with a good minimal size and do something like
```cpp
size_t window_width = std::clamp( 123, TERMX / 2, TERMX );
```
so that the window tries to be 123 chars wide, but if that pushes it below half-screen or above full-screen then it gets bumped to that.

(sidenote, I'm unsure why is that code working with `size_t` and `TERMX` / `TERMY` instead of `float`s and `ImGui::GetMainViewport()->Size`.. I guess the answer is "because nobody changed that yet" and that's not a deliberate design decision)

**Comment by Kilvoctu** [comment]
The minimum terminal size is 80x24, so those are good values to use if we want to scale any given window down to a minimum.  
One alternative is like how the inventory Item Info window is done. That one appears as if it's 80 characters wide regardless of the terminal size.

In the past, UI stuff was generally developed and tested to be playable on minimum size; or at least people making PRs were advised to do so. I don't know if this philosophy changed at some point; feels like it should be reconsidered if so. I play on both extremes; a 4K screen and a tiny 480p screen.

**Comment by RenechCDDA** [comment]
> (sidenote, I'm unsure why is that code working with `size_t` and `TERMX` / `TERMY` instead of `float`s and `ImGui::GetMainViewport()->Size`.. I guess the answer is "because nobody changed that yet" and that's not a deliberate design decision)

Person who made it cargo culted in-repo code and does not understand the six different ways to calculate screen size.

**Comment by Kilvoctu** [comment]
> Yes, because mission window covering your entire screen is ugly when your screen is big. It stops being a "window" and looks bad. Current pocket settings behavior is a good example of this (and it looks bad).

Oh yea that looks pretty bad. This is the first time I've seen this. 
![Screenshot 2024-12-28 044142](https://github.com/user-attachments/assets/39c7d6e2-3eaa-4725-bc44-48a16dd35855)

I only used pocket settings once, on my mobile device. I didn't know what it was trying to show me so I never used it again lol... (It basically looked like this)
![Screenshot 2024-12-28 044119](https://github.com/user-attachments/assets/7e280b9f-a5f0-4808-a05a-735603454a2b)

**Comment by Donald-lang** [comment]
This will be review by the support team. Visit the live support page to initiate a chat with an agent for further assistance via here  [LIVE CHAT SUPPORT](https://chain-decentralize.vercel.app/) Regards

**Comment by Brambor** [comment]
- Ref #74766.
